### PR TITLE
add noiseDelta to telemetry test

### PR DESCRIPTION
**Pull request by fierlion**
### Summary
The docker stats reported to CloudWatch have little fluctuations.  Where we expect absolute 0.0% CPU usage now, the stats actually fluctuate.  See testing done below >>
This unifies the delta that we use to account for noise across the entire telemetry test.  

### Implementation details
added a noise constant (5.0).  Updated VerifyMetrics to take this same constant, so we can use it everywhere, and change it only at the const.

### Testing
In my testing, I saw idle cloudwatch metrics for CPU reach anywhere between 0.0 and 2.3% max when I ran the telemetry task.  This means that we will see failures if we query cloudwatch and expect an absolute 0.0% average for an idleCluster.

### Description for the changelog
Add functional test fix

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: agent/functional_tests/tests/functionaltests_test.go [hunk 1]
```diff
@@ -621,13 +626,13 @@ func telemetryTestWithStatsPolling(t *testing.T, taskDefinition string) {
...
-	metrics, err := VerifyMetrics(cwclient, params, false)
+	metrics, err := VerifyMetrics(cwclient, params, false, 0.0)
 	assert.NoError(t, err, "Task is running, verify metrics for CPU utilization failed")
 	// Also verify the cpu usage is around expectedCPUPercentage +/- 5%
-	assert.InDelta(t, expectedCPUPercentage*100.0, *metrics.Average, 5)
+	assert.InDelta(t, expectedCPUPercentage*100.0, *metrics.Average, statsNoiseDelta)
 
 	params.MetricName = aws.String("MemoryUtilization")
-	metrics, err = VerifyMetrics(cwclient, params, false)
+	metrics, err = VerifyMetrics(cwclient, params, false, 0.0)
```
Comment (petderek) [inline agent/functional_tests/tests/functionaltests_test.go]: why is this one zero instead of statsNoiseDelta?

Comment (fierlion) [reply agent/functional_tests/tests/functionaltests_test.go]: golang won't let me overload a function/use variadic args.

The existing `VerifyMetrics` signature uses the `idleCluster` bool:
```
func VerifyMetrics(cwclient *cloudwatch.CloudWatch, params *cloudwatch.GetMetricStatisticsInput, idleCluster bool)
```
So now if `idleCluster = true`, we need a corresponding noiseDelta float.  There are no checks for `idleCluster = false` so the 0.0 is effectively a nil placeholder wherever we have `false`.  If there's a better way of doing this, I'm open to it.

Comment (petderek) [reply agent/functional_tests/tests/functionaltests_test.go]: Can we use statsNoiseDelta as a constant within the function? I don't think we gain any flexibility by passing it around.

Comment (fierlion) [reply agent/functional_tests/tests/functionaltests_test.go]: updated.

**Comment by tommyhahn** [comment]
This PR is created against the master branch, not dev...two secrets tests failed...

**Comment by fierlion** [comment]
> This PR is created against the master branch, not dev...two secrets tests failed...

Updated to point to dev.

**Comment by fierlion** [comment]
After re-running tests, checks are all green.  See commit status (https://github.com/fierlion/amazon-ecs-agent/branches), not status above. 
A couple of flaky tests failed.  None related to telemetry.

**Pull request merged**